<?xml version="1.0"?>
<launch>
	<arg name="rosbag_file" default="" />
	<arg name="scan_topic" default="/scan" />
	<param name="use_sim_time" value="true" />
	<arg name="rosbag_start_delay" default="1.0" />
	<arg name="npy_directory" default="$(find data_labeling)/npy_files/"/>

	<node pkg="petra" type="petra" name="petra" args="" output="screen">
    	<rosparam file="$(find petra)/config/parameters.yaml" command="load" />
  	</node>

	<node pkg="data_labeling" type="data_labeling_with_petra" name="data_labeling_with_petra" output="screen">
		<param name="rosbag_file" value="$(arg rosbag_file)" />
		<param name="scan_topic" value="$(arg scan_topic)" />
    	<param name="npy_directory" value="$(arg npy_directory)" />
	</node>


	<node pkg="rosbag" type="play" name="player" output="screen" args=" --clock $(arg rosbag_file) /rb1_base/front_laser/scan:=/scan" required="true"
    launch-prefix="bash -c 'sleep $(arg rosbag_start_delay); $0 $@' "/>

</launch>
